File: 2017 - 2/Programação de Computadores/trabalho java/src/Pessoa.java
/**
 *
@author João Vitor Azevedo Marciano - 743554
 * 
 * Proposta: 
 * Implemente a classe Pessoa que possua como atributos: nome, peso 
 * (em quilogramas)
e altura (em metros). Inclua na classe um construtor para definir os atributos 
* e os seguintes métodos:
– Métodos (getters e setters) para ler e alterar cada um dos atributos em 
* separado.
– O método double getIMC() que calcula o índice de Massa Corporal (IMC). 
* O cálculo é feito através da fórmula IMC = peso * altura2 .
– O método void imprimeCategoria() que informa a faixa de categoria do IMC 
* que a pessoa se encontra, utilizando a tabela adequada
 */
public class Pessoa {
    private String nome;
    private float pesoKg;
    private float altura;

    // Construtor
    public Pessoa(String nome, float pesoKg, float altura) {
        this.nome = nome;
        this.pesoKg = pesoKg;
        this.altura = altura;
    }
    //Calcula e retorna o IMC da pessoa
    public double getIMC(){
        return (this.pesoKg / (altura * altura));
    }
    //Imprime a categoria à qual tal pessoa pertence, baseado em seu IMC
    public void imprimeCategoria(){
        if(this.getIMC() < 18.5)
            System.out.println("Abaixo do Peso");        
        
        else if(this.getIMC() < 25)
            System.out.println("Peso Normal");
        
        else if (this.getIMC() < 30) 
            System.out.println("Sobrepeso");
        
        else if(this.getIMC() < 35)
            System.out.println("Obesidade grau I");
        
        else if(this.getIMC() < 40) 
            System.out.println("Obesidade grau II");
                
        else 
            System.out.println("Obesidade grau III e seu IMC é " + this.getIMC());
        
              
        
    }
    
    
    
    
    
    //Getters
    
    public String getNome() {
        return this.nome;
    }
    
    public float getAltura() {
        return this.altura;
    }

    public float getPesoKg() {
        return this.pesoKg;
    }

    //Setters
    public void setAltura(float altura) {
        this.altura = altura;
    }

    public void setPesoKg(float pesoKg) {
        this.pesoKg = pesoKg;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }
    
    
    
    
    
    
    
    
}

File: 2017 - 2/Programação de Computadores/Trabalho01-PC-CplusPlus/src/Pessoa.cpp
#include "Pessoa.h"
#include <string.h>
#include <iostream>

using namespace std;

Pessoa::Pessoa()
{

}
Pessoa::Pessoa(char name[50], double pesin, double height)
{
    strcpy(nome,name); // nome = name;
    pesoKg = pesin;
    altura = height;
}



double Pessoa::getIMC(double pesinho, double alturinha){
        return (pesinho / (alturinha * alturinha));


}


void Pessoa::imprimeCategoria(){

        if(getIMC(pesoKg,altura) < 18.5 ){
            cout << "Abaixo do peso" << endl;
            }

        else if(getIMC(pesoKg, altura) < 25 && getIMC(pesoKg, altura) >= 18.5)
        cout << "Peso normal " << endl;


        else if(getIMC(pesoKg, altura) >= 25 && getIMC(pesoKg, altura) < 30)
            cout << "Sobrepeso" << endl;

        else if(getIMC(pesoKg, altura) >= 30 && getIMC(pesoKg, altura)< 35)
            cout << "Obesidade grau I" << endl;

        else if(getIMC(pesoKg, altura) >=35 && getIMC(pesoKg, altura) < 40)
            cout << "Obesidade grau II "  << endl;

        else if(getIMC(pesoKg, altura) >= 40)
            cout << "Obesidade grau III" << endl;

}

//getters
double Pessoa::getAltura(){
    return altura;

}
double Pessoa::getPeso(){
    return pesoKg;

}
char Pessoa::getNome(){
    return *nome; // não consigo implementar esse, problemas gerenciando "char" .

}

// setters
void Pessoa::setAltura(double value){
    altura = value;

}

void Pessoa::setPeso(double value){
    pesoKg = value;

}

void Pessoa::setNome(char noma[50]){
    strcpy(nome, noma);

}
